hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/handler/CloseRegionHandler.java

```diff
@@ -117,9 +117,7 @@ public void process() throws IOException {
         HConstants.NO_SEQNUM, Procedure.NO_PROC_ID, -1, regionInfo));
 
       // Done!  Region is closed on this RS
-      this.rsServices.getRegionsInTransitionInRS().
-        remove(this.regionInfo.getEncodedNameAsBytes(), Boolean.FALSE);
-      LOG.debug("Closed {}" + region.getRegionInfo().getRegionNameAsString());
```

Comment:
Hmm.. Is it the finally that is wrong?

Here we remove only if value matches FALSE. The finally removes whatever the value.

But I suppose no one else can add an entry while there is one in the map so the finally remove should be fine?

Reply:
@saintstack In this specific context, remove only if value matches FALSE or remove anyway is same. It enters in the loop with value FALSE and it wont be changed to TRUE(going through the code).

Reply:
Test failures are all due to timeout, I run the failed test cases locally with adhoc_run_tests.sh locally, it passed for me.